lib/puppet/pops/evaluator/runtime3_support.rb

```diff
@@ -266,6 +266,14 @@ def external_call_function(name, args, scope, &block)
   end
 
   def call_function(name, args, o, scope, &block)
+    if args.size >= 1
+      receiver_type = Types::TypeCalculator.infer(args[0])
+      if receiver_type.is_a?(Types::PObjectType)
+        member = receiver_type[name]
+        return member.invoke(scope, args, &block) unless member.nil?
+      end
+    end
```

Comment:
As per the specification - this functionality should be in the evaluator where it can differentiate between a method call and a function call.